src/routes/notifications/handlers.rs

```diff
@@ -51,33 +59,46 @@ pub async fn post_registration(
         requests.push((&safe_registration.chain_id, client.post(request)));
     }
 
-    let (error_chain_ids, error_body) = {
-        let mut error_chain_ids: Vec<&str> = vec![];
-        let mut errors: Vec<Value> = vec![];
-        for (chain_id, request) in requests.into_iter() {
-            match request.await {
-                Err(api_error) => {
-                    error_chain_ids.push(chain_id);
-                    errors.push(json!({
-                        chain_id :   RawValue::from_string(api_error.details.message.unwrap_or(String::from("Unknown notification registration issue")))?
-                    }))
-                }
-                _ => {}
+    let mut errors: Vec<NotificationRegistrationError> = vec![];
+    for (chain_id, request) in requests.into_iter() {
+        match request.await {
+            Err(api_error) => {
+                let notification_registration_error = NotificationRegistrationError {
+                    status_code: api_error.status,
+                    chain_id: String::from(chain_id),
+                    error: json!(
+                            {chain_id :   RawValue::from_string(api_error.details.message.unwrap_or(String::from("Unknown notification registration issue")))?}),
+                };
+                errors.push(notification_registration_error);
             }
+            _ => {}
         }
-        (error_chain_ids, json!(errors))
-    };
+    }
 
-    if error_chain_ids.is_empty() {
+    if errors.is_empty() {
         Ok(())
     } else {
-        Err(ApiError::new_from_message_with_debug(
-            format!(
-                "Push notification registration failed for chain IDs: {}",
-                error_chain_ids.join(", ")
-            ),
-            Some(error_body),
-        ))
+        let has_server_error = errors
+            .iter()
+            .any(|error| (500..600).contains(&error.status_code));
```

Comment:
> We have something similar in our cache: https://github.com/gnosis/safe-client-gateway/blob/main/src/cache/cache_op_executors.rs#L57
> Maybe we should create some utils for this.

I've looked into this since we do this in more places. In my opinion this should not be part of this ticket.

> Also do you create an array with 100 elements here for every entry of the errors array?

Not exactly sure what do you mean by this. If it's related with the `[500, 600[` then it's a range – https://doc.rust-lang.org/src/core/ops/range.rs.html#80-87 (no arrays involved)

Reply:
> I've looked into this since we do this in more places. In my opinion this should not be part of this ticket.

In this case I would already open a follow up PR where I fix this (or at least an issue) else we will build up technical depth. While I agree that a PR should try to stick to the minimal changes we need to keep in mind that if we need to keep technical depth low.

Reply:
I think even if having multiple tickets is the right way to do some stuff we lose a lot of time that way. Sometimes refactors make sense when implementing/changing something, as long as it's well tested, to keep technical depth low as @rmeissner says

Reply:
I think it depends. If we bundle every change here and we want to revert this change because it's not working as intended we would be reverting a refactor too.

Reply:
Yes, agree. I think it goes case by case, but what I meant is that we shouldn't be so strict on "this change does not belong to this PR" 😉